jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,2724 @@
+			if (result) {
+				return intAtIndex;
+			}
+		}
+	}
+
+	public final int getAndBitwiseAndIntRelease(Object obj, long index, int value) {
+		for (;;) {
+			int intAtIndex = getInt(obj, index);
+			boolean result = weakCompareAndSetIntRelease(obj, index, intAtIndex, intAtIndex & value);
+			if (result) {
+				return intAtIndex;
+			}
+		}
+	}
+
+	public final int getAndBitwiseAndIntAcquire(Object obj, long index, int value) {
+		for (;;) {
+			int intAtIndex = getInt(obj, index);
+			boolean result = weakCompareAndSetIntAcquire(obj, index, intAtIndex, intAtIndex & value);
+			if (result) {
+				return intAtIndex;
+			}
+		}
+	}
+
+	public final int getAndBitwiseXorInt(Object obj, long index, int value) {
+		for (;;) {
+			int intAtIndex = getIntVolatile(obj, index);
+			boolean result = weakCompareAndSetInt(obj, index, intAtIndex, intAtIndex ^ value);
+			if (result) {
+				return intAtIndex;
+			}
+		}
+	}
+
+	public final int getAndBitwiseXorIntRelease(Object obj, long index, int value) {
+		for (;;) {
+			int intAtIndex = getInt(obj, index);
+			boolean result = weakCompareAndSetIntRelease(obj, index, intAtIndex, intAtIndex ^ value);
+			if (result) {
+				return intAtIndex;
+			}
+		}
+	}
+
+	public final int getAndBitwiseXorIntAcquire(Object obj, long index, int value) {
+		for (;;) {
+			int intAtIndex = getInt(obj, index);
+			boolean result = weakCompareAndSetIntAcquire(obj, index, intAtIndex, intAtIndex ^ value);
+			if (result) {
+				return intAtIndex;
+			}
+		}
+	}
+
+	public final long getAndBitwiseOrLong(Object obj, long index, long value) {
+		for (;;) {
+			long longAtIndex = getLongVolatile(obj, index);
+			boolean result = weakCompareAndSetLong(obj, index, longAtIndex, longAtIndex | value);
+			if (result) {
+				return longAtIndex;
+			}
+		}
+	}
+
+	public final long getAndBitwiseOrLongRelease(Object obj, long index, long value) {
+		for (;;) {
+			long longAtIndex = getLong(obj, index);
+			boolean result = weakCompareAndSetLongRelease(obj, index, longAtIndex, longAtIndex | value);
+			if (result) {
+				return longAtIndex;
+			}
+		}
+	}
+
+	public final long getAndBitwiseOrLongAcquire(Object obj, long index, long value) {
+		for (;;) {
+			long longAtIndex = getLong(obj, index);
+			boolean result = weakCompareAndSetLongAcquire(obj, index, longAtIndex, longAtIndex | value);
+			if (result) {
+				return longAtIndex;
+			}
+		}
+	}
+
+	public final long getAndBitwiseAndLong(Object obj, long index, long value) {
+		for (;;) {
+			long longAtIndex = getLongVolatile(obj, index);
+			boolean result = weakCompareAndSetLong(obj, index, longAtIndex, longAtIndex & value);
+			if (result) {
+				return longAtIndex;
+			}
+		}
+	}
+
+	public final long getAndBitwiseAndLongRelease(Object obj, long index, long value) {
+		for (;;) {
+			long longAtIndex = getLong(obj, index);
+			boolean result = weakCompareAndSetLongRelease(obj, index, longAtIndex, longAtIndex & value);
+			if (result) {
+				return longAtIndex;
+			
```

Comment:
Exceptions with messages should use com.ibm.oti.util.Msg.getString() for national language support (NLS).